src/datatype.c

```diff
@@ -113,6 +113,55 @@ jl_datatype_t *jl_new_uninitialized_datatype(void)
     return t;
 }
 
+#include "support/htable.h"
+#include "support/htable.inc"
+
+static uint32_t _hash_djb2(uint32_t hash, const char* mem, size_t s) {
```

Comment:
style nits
```suggestion
static uint32_t _hash_djb2(uint32_t hash, const char *mem, size_t s)
{
```

Reply:
We can probably use `memhash_seed` / `memhash32_seed` here, which is a lot slower, but does some more bit mixing